packages/website/ts/pages/shared/markdown_code_block.tsx

```diff
@@ -7,14 +7,23 @@ interface MarkdownCodeBlockProps {
     language: string;
 }
 
-export function MarkdownCodeBlock(props: MarkdownCodeBlockProps) {
-    return (
-        <span style={{fontSize: 16}}>
-            <HighLight
-                className={props.language || 'js'}
-            >
-                {props.literal}
-            </HighLight>
-        </span>
-    );
+interface MarkdownCodeBlockState {}
+
+export class MarkdownCodeBlock extends React.Component<MarkdownCodeBlockProps, MarkdownCodeBlockState> {
```

Comment:
Why not `React.PureComponent`? it will give the same behaviour.

Reply:
Turns out you can only do this when all sub-components are also pure. This is not the case of `HighLight` component.